task_3/counter.cpp

```diff
@@ -0,0 +1,324 @@
+			if (found1 < found3 || found2 < found3)	
+				s.erase (found3, s.size() - found3);
+			else {	
+				s.erase (found3, s.size() - found3);
+				s = s.size() > 2 ? "u//" : "//";
+			}
+
+		if (found1 != std::string::npos && found2 == std::string::npos && found3 != std::string::npos){
+			if (found1 < found3)
+				s.erase (found3, s.size() - found3);
+			else {
+				s.erase (found3, s.size() - found3);
+				s = s.size() > 2 ? "u//" : "//";
+			}
+		}
+		
+		if (found1 == std::string::npos && found2 != std::string::npos && found3 != std::string::npos){
+			if (found2 < found3)
+				s.erase (found3, s.size() - found3);
+			else {
+				s.erase (found3, s.size() - found3);
+				s = s.size() > 2 ? "u//" : "//";
+			}
+		}
+
+		s.erase (remove_if(s.begin(), s.end(), isspace), s.end());
+		if (!s.size())
+			return;
+
+		// check if multiline and count number off comment blocks
+		int n1 = 0;
+		int n2 = 0;
+		std::size_t found11;
+
+		found1 = s.find("/*");
+		found2 = s.find("*/");
+		bool nextLineOfMultilineComment {found1 > found2 ? true : false};
+
+		while (found1 != std::string::npos) {
+			++n1;
+			found11 = found1;
+			found1 = s.find("/*", found1 + 2);
+		}
+
+		while (found2 != std::string::npos) {
+			++n2;
+			found2 = s.find("*/", found2 + 2);
+		}
+
+		if (!nextLineOfMultilineComment) {
+			bool multiline {n1 != n2 ? true : false};	
+			if (multiline)	{	
+				s.erase (found11 + 2, s.size() - found11 - 2);	// erase last comment after /*
+				found1 = s.find_first_not_of("/*");
+				if (found1 != std::string::npos)
+					s = "c/*";
+				else
+					s = "/*";
+			} else {
+				found1 = s.find_first_not_of("/*");
+				if (found1 != std::string::npos)
+					s = "c//";
+				else
+					s = "//";
+			}
+		} else {
+			bool multiline {n1 == n2 ? true : false};
+			if (multiline)	{
+				s.erase (found11 + 2, s.size() - found11 - 2);	// erase last comment after /*
+				found1 = s.find_first_not_of("/*");
+				if (found1 != std::string::npos)
+					s = "c/*";
+				else
+					s = "/*";
+			} else {
+				found1 = s.find_first_not_of("/*");
+				if (found1 != std::string::npos)
+					s = "c//";
+				else
+					s = "//";
+			}
+		}
+		//cout << "string out:" << s << ":" << endl;
+	}
+	void Counter::processFile(path & pathCurr, Statistics & stat) {
+		std::ifstream file;
+		file.open (pathCurr, std::ofstream::out);
+		if ( (file.rdstate() & std::ifstream::failbit ) != 0 ) {
+			global_stream_lock.lock();
+			std::cerr << "[" << boost::this_thread::get_id() << "] couldn't open file <" << pathCurr 
+				<< ">. Skipping." << endl;
+			global_stream_lock.unlock();
+			return;
+		}
+
+		string s;
+		bool inMultiline = false;
+		for (size_t line {1}; getline(file, s); ++line) {
+			//global_stream_lock.lock();cout << "string read:" << s << ":." << endl;global_stream_lock.unlock();
+			simplify(s);
+			//global_stream_lock.lock();cout << "string simplify:" << s << ":." << endl;global_stream_lock.unlock();
+
+			++stat.linesTotal;
+			
+			if (s.compare("b")) {
```

Comment:
You have hardcoded signal sequences **"b"** or **"c//"** etc. This is uncorrect, because the signal can be misread if the program' code contain one of the signal lines. The best solution here is to previously replace every signal sequence with something different. E.g. you use **"c//"** and **"u//"**: then replace previously **"c//"** or **"u//"** to **"i//"** in the code (Note the interpretation will not suffer).

Reply:
in this stage no more original string. only marks like "b" - means line was blank, "c" - means line with code, "s' - means single line comment, etc